### PR TITLE
pyretis feedstock?

### PR DESCRIPTION
**Pull request by EnricoRiccardi**
this is what I am missing to have our code (pyretis) to conda?

thanks

### CODE REVIEW

**Comment by EnricoRiccardi** [comment]
retry

**Comment by jjhelmus** [comment]
@EnricoRiccardi This pull request has 695 files added to this repository which I believe is causing some issues. Is there a reason why you need to include all these files?  Typically adding a single `meta.yaml` files to a subdirectory in the recipes folder is all that is needed. All of these files also makes it very difficult to review this PR as the `meta.yaml` cannot be viewed though the GitHub website (at least I cannot).

**Comment by EnricoRiccardi** [comment]
@jjhelmus thanks a lot first for you answer and trying to help me. I took away all the non necessary files. My impression is that there must be something extremely obvious that I don't get and that block everything.

I tried many combos to try to make it work, moved the code from gitlab got github, changed metafile, changed the position of the metafile... dunno what to try... always getting the same error..

File: pyretis/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#{% set version = "0.9.2" %}
+#{% set sha256 = "f738210a323dd8f84f98f33f0836dfdcc04f3fa8559a4e847f8eb311f147c7cc" %}
+
+package:
+   name: pyretis
+   version: 0.9.2
+#  name: {{ name|lower }}
+#  version: {{ version }}
+
+source:
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: Is it possible to use the source code from pypi.io (0.9.2.dev2)? See:

https://conda-forge.org/docs/meta.html#build-from-tarballs-not-repos

File: pyretis/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#  git_url: https://github.com/EnricoRiccardi/PyRETIS-on-GIT/tree/master/pyretis
+#  git_url: https://github.com/EnricoRiccardi/pyretis/edit/master/recipes/pyretis/pyretis
+
+  sha256: {{ sha256 }}
+
+build:
+#  preserve_egg_dir: True
+#  number: 0
+#  skip: True  # [ not py2k or win #]
+  script: python setup.py install
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: It looks like the `setup.py` uses `setuptools` so try out:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L26

File: pyretis/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  sha256: {{ sha256 }}
+
+build:
+#  preserve_egg_dir: True
+#  number: 0
+#  skip: True  # [ not py2k or win #]
+  script: python setup.py install 
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: If you use the `--single-version-externally-managed` flag in the `script` entry mentioned above, you should only need `python` and `setuptools` as build requirements.

File: pyretis/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - numpy
+    - scipy
+    - pandas
+#    - mdtraj 
+    - matplotlib
+    - jinja2
+    - tqdm
+    - colorama
+
+  run:
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: If any of the version restrictions are important, please add them:

https://gitlab.com/pyretis/pyretis/blob/master/setup.py#L74-80

File: pyretis/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - tqdm
+    - colorama
+
+test:
+  imports:
+    - pyretis
+
+about:
+  home: http://www.pyretis.org
+  license: LGPL 2.1 or later
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: Please include a `license_file`. Also take a look at the example for other fields to populate:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L44-L65

File: pyretis/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+summary: ''
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: You definitely need to add yourself as a maintainer. See:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L67-L72

File: pyretis/meta.yaml (hunk 4 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: not yet, I still need to be able to upload the first version

File: pyretis/meta.yaml (hunk 5 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: I am afraid to add at this moment other error sources. I would like to first understand what is going wrong.  I am starting to consider to not use conda any longer, I am loosing too much time and other people, like you, too... :(

File: pyretis/meta.yaml (hunk 1 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: I tried. Still no linting

File: pyretis/meta.yaml (hunk 3 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: if I do this, wouldn't I loose all the conda checks for package compatibility? Shall I take out setuptool from setup.py?

File: pyretis/meta.yaml (hunk 6 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: done

File: pyretis/meta.yaml (hunk 1 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: ok, seems that I finally managed to make it work. Thank you tons. 

I would like to add mdtraj in the list of packages, but it fails if I do so. Do you have any tips for that?

Once more, really thanks!

File: pyretis/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py<3]
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  ## preserve_egg_dir: True
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: It looks like only `python` and `setuptools` are required to build the package (the `--single-version-externally-managed` flag, means that `install_requires` is ignored).

File: pyretis/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  - jinja2
+
+test:
+  imports:
+  - pyretis
+
+about:
+  home: http://www.pyretis.org
+  license: LGPL-2.1 or later 
+  summary: 'A rare event code based for Transition Interface Sampling'
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: Could you add info for `license_file`, and `dev_url` to the `about` section?

File: pyretis/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  - tqdm
+  - future
+  - numpy
+  - scipy
+  - matplotlib
+  - jinja2
+
+test:
+  imports:
+  - pyretis
```
Comment (synapticarbors) [inline pyretis/meta.yaml]: If any of the command line scripts have some sort of `-h`/`--help` flag that you could use to test to make sure those are in place, please add them as in:

https://conda.io/docs/building/meta-yaml.html#test-commands

**Comment by jjhelmus** [comment]
Before merging this PR these commits should be squashed.  If they are not this will add a lot of bulk to the repository which everyone who forks will need to download.

**Comment by EnricoRiccardi** [comment]
Erased the commit history

**Pull request closed without merging**